Errors reported by mirage 4.5.0

I did some modification, namely Mirage_logs no longer being a functor. Now, I ran mirage configure, expecting the build to complain.

The error I got is:
```
$ gmake build
dune build --profile release --root . ./dist
File "lib/devices/runtime_arg.ml", line 161, characters 46-52:
Error: Unbound module Pclock
gmake: *** [Makefile:56: build] Error 1
```

It is not obvious to me, where I can find `lib/devices/runtime_arg.ml`

--- a/lib/functoria/engine.ml
+++ b/lib/functoria/engine.ml
@@ -163,21 +163,6 @@ let append_main main msg fmt =
           Fmt.pf ppf "%s@." str))
     fmt
 
-let configure info t =
-  let* main = main info in
-  let f (v : t) =
-    let (D { dev; args; _ }) = v in
-    let* () = Device.configure dev info in
-    if args = [] then Action.ok ()
-    else
-      append_main main "configure" "module %s = %a\n" (Device.Graph.impl_name v)
-        module_expression (dev, args)
-  in
-  iter_actions f t
-
-let meta_init fmt (connect_name, result_name) =
-  Fmt.pf fmt "  let _%s = Lazy.force %s in@ " result_name connect_name
-
 let pp_pos ppf = function
   | None -> ()
   | Some (file, line, _, _) -> Fmt.pf ppf "# %d %S@." line file
@@ -185,6 +170,22 @@ let pp_pos ppf = function
 let reset_pos { dir; path; lines } =
   let file = Fpath.(dir // path) |> Fpath.normalize |> Fpath.to_string in
   Some (file, lines + 1, 0, 0)
+
+let configure info t =
+  let* main = main info in
+  let f (v : t) =
+    let (D { dev; args; _ }) = v in
+    let* () = Device.configure dev info in
+    if args = [] then Action.ok ()
+    else
+      let* () = append_main main "reset" "%a" pp_pos (reset_pos main) in
+      append_main main "configure" "module %s = %a\n" (Device.Graph.impl_name v)
+        module_expression (dev, args)
+  in
+  iter_actions f t
+
+let meta_init fmt (connect_name, result_name) =
+  Fmt.pf fmt "  let _%s = Lazy.force %s in@ " result_name connect_name
 
 let emit_connect fmt (iname, names, runtime_args, connect_code) =
   (* We avoid potential collision between double application

--- a/lib/functoria/engine.ml
+++ b/lib/functoria/engine.ml
@@ -172,8 +172,8 @@ let reset_pos { dir; path; lines } =
   Some (file, lines + 1, 0, 0)
 
 let configure info t =
-  let* main = main info in
   let f (v : t) =
+    let* main = main info in
     let (D { dev; args; _ }) = v in
     let* () = Device.configure dev info in
     if args = [] then Action.ok ()

--- a/test/functoria/e2e/errors.t
+++ b/test/functoria/e2e/errors.t
@@ -14,14 +14,14 @@ not the right ones. First, too many parameters:
 
   $ ./test.exe configure -f errors/in_functor_too_many.ml
   $ dune build 2>&1 | head -n1 | cut -d',' -f'-2'
-  File "errors/test/main.ml", line 8
+  File "errors/test/main.ml", line 9
   $ ./test.exe clean -f errors/in_functor_too_many.ml
 
 Then, not enough:
 
   $ ./test.exe configure -f errors/in_functor_not_enough.ml
   $ dune build
-  File "errors/test/main.ml", line 31, characters 3-26:
+  File "errors/test/main.ml", line 33, characters 3-26:
   Error: The module Unikernel_make__4 is a functor, it cannot have any components
   [1]
   $ ./test.exe clean -f errors/in_functor_not_enough.ml

--- a/test/mirage/random/run.t
+++ b/test/mirage/random/run.t
@@ -96,9 +96,15 @@ Configure the project for Unix:
     Mirage_runtime.logs
   ;;
   
+  # 77 "mirage/main.ml"
+  
   module Mirage_logs_make__8 = Mirage_logs.Make(Pclock)
   
+  # 81 "mirage/main.ml"
+  
   module Mirage_crypto_rng_mirage_make__11 = Mirage_crypto_rng_mirage.Make(Unix_os.Time)(Mclock)
+  
+  # 85 "mirage/main.ml"
   
   module App_make__12 = App.Make(Mirage_crypto_rng_mirage_make__11)
   
@@ -106,7 +112,7 @@ Configure the project for Unix:
   # 7 "lib/devices/argv.ml"
     Bootvar.argv ()
   );;
-  # 87 "mirage/main.ml"
+  # 93 "mirage/main.ml"
   
   let struct_end__2 = lazy (
     let __bootvar__1 = Lazy.force bootvar__1 in
@@ -114,21 +120,21 @@ Configure the project for Unix:
   # 47 "lib/functoria/job.ml"
     return Mirage_runtime.(with_argv (runtime_args ()) "random" _bootvar__1)
   );;
-  # 95 "mirage/main.ml"
+  # 101 "mirage/main.ml"
   
   let printexc__3 = lazy (
     let _backtrace = (backtrace__key ()) in
   # 397 "lib/mirage.ml"
     return (Printexc.record_backtrace _backtrace)
   );;
-  # 102 "mirage/main.ml"
+  # 108 "mirage/main.ml"
   
   let hashtbl__4 = lazy (
     let _randomize_hashtables = (randomize_hashtables__key ()) in
   # 406 "lib/mirage.ml"
     return (if _randomize_hashtables then Hashtbl.randomize ())
   );;
-  # 109 "mirage/main.ml"
+  # 115 "mirage/main.ml"
   
   let gc__5 = lazy (
     let _allocation_policy = (allocation_policy__key ()) in
@@ -157,19 +163,19 @@ Configure the project for Unix:
     custom_minor_max_size = (match _custom_minor_max_size with None -> ctrl.custom_minor_max_size | Some x -> x) })
   )
   );;
-  # 138 "mirage/main.ml"
+  # 144 "mirage/main.ml"
   
   let mirage_runtime__6 = lazy (
     let _delay = (delay__key ()) in
   # 302 "lib/mirage.ml"
     Unix_os.Time.sleep_ns (Duration.of_sec _delay)
   );;
-  # 145 "mirage/main.ml"
+  # 151 "mirage/main.ml"
   
   let pclock__7 = lazy (
     return ()
   );;
-  # 150 "mirage/main.ml"
+  # 156 "mirage/main.ml"
   
   let mirage_logs_make__8 = lazy (
     let __pclock__7 = Lazy.force pclock__7 in
@@ -181,17 +187,17 @@ Configure the project for Unix:
     Logs.set_reporter reporter;
     Lwt.return reporter
   );;
-  # 162 "mirage/main.ml"
+  # 168 "mirage/main.ml"
   
   let unix_os_time__9 = lazy (
     return ()
   );;
-  # 167 "mirage/main.ml"
+  # 173 "mirage/main.ml"
   
   let mclock__10 = lazy (
     return ()
   );;
-  # 172 "mirage/main.ml"
+  # 178 "mirage/main.ml"
   
   let mirage_crypto_rng_mirage_make__11 = lazy (
     let __unix_os_time__9 = Lazy.force unix_os_time__9 in
@@ -201,7 +207,7 @@ Configure the project for Unix:
   # 18 "lib/devices/random.ml"
     Mirage_crypto_rng_mirage_make__11.initialize (module Mirage_crypto_rng.Fortuna)
   );;
-  # 182 "mirage/main.ml"
+  # 188 "mirage/main.ml"
   
   let app_make__12 = lazy (
     let __mirage_crypto_rng_mirage_make__11 = Lazy.force mirage_crypto_rng_mirage_make__11 in
@@ -209,7 +215,7 @@ Configure the project for Unix:
   # 3 "config.ml"
     (App_make__12.start _mirage_crypto_rng_mirage_make__11 : unit io)
   );;
-  # 190 "mirage/main.ml"
+  # 196 "mirage/main.ml"
   
   let mirage_runtime__13 = lazy (
     let __struct_end__2 = Lazy.force struct_end__2 in
@@ -229,7 +235,7 @@ Configure the project for Unix:
   # 385 "lib/mirage.ml"
     return ()
   );;
-  # 210 "mirage/main.ml"
+  # 216 "mirage/main.ml"
   
   let () =
     let t = Lazy.force struct_end__2 >>= fun _ ->
@@ -348,9 +354,15 @@ Configure the project for Xen:
     Mirage_runtime.logs
   ;;
   
+  # 77 "mirage/main.ml"
+  
   module Mirage_logs_make__8 = Mirage_logs.Make(Pclock)
   
+  # 81 "mirage/main.ml"
+  
   module Mirage_crypto_rng_mirage_make__11 = Mirage_crypto_rng_mirage.Make(Xen_os.Time)(Mclock)
+  
+  # 85 "mirage/main.ml"
   
   module App_make__12 = App.Make(Mirage_crypto_rng_mirage_make__11)
   
@@ -358,7 +370,7 @@ Configure the project for Xen:
   # 21 "lib/devices/argv.ml"
     Bootvar.argv ()
   );;
-  # 87 "mirage/main.ml"
+  # 93 "mirage/main.ml"
   
   let struct_end__2 = lazy (
     let __bootvar__1 = Lazy.force bootvar__1 in
@@ -366,21 +378,21 @@ Configure the project for Xen:
   # 47 "lib/functoria/job.ml"
     return Mirage_runtime.(with_argv (runtime_args ()) "random" _bootvar__1)
   );;
-  # 95 "mirage/main.ml"
+  # 101 "mirage/main.ml"
   
   let printexc__3 = lazy (
     let _backtrace = (backtrace__key ()) in
   # 397 "lib/mirage.ml"
     return (Printexc.record_backtrace _backtrace)
   );;
-  # 102 "mirage/main.ml"
+  # 108 "mirage/main.ml"
   
   let hashtbl__4 = lazy (
     let _randomize_hashtables = (randomize_hashtables__key ()) in
   # 406 "lib/mirage.ml"
     return (if _randomize_hashtables then Hashtbl.randomize ())
   );;
-  # 109 "mirage/main.ml"
+  # 115 "mirage/main.ml"
   
   let gc__5 = lazy (
     let _allocation_policy = (allocation_policy__key ()) in
@@ -409,19 +421,19 @@ Configure the project for Xen:
     custom_minor_max_size = (match _custom_minor_max_size with None -> ctrl.custom_minor_max_size | Some x -> x) })
   )
   );;
-  # 138 "mirage/main.ml"
+  # 144 "mirage/main.ml"
   
   let mirage_runtime__6 = lazy (
     let _delay = (delay__key ()) in
   # 302 "lib/mirage.ml"
     Xen_os.Time.sleep_ns (Duration.of_sec _delay)
   );;
-  # 145 "mirage/main.ml"
+  # 151 "mirage/main.ml"
   
   let pclock__7 = lazy (
     return ()
   );;
-  # 150 "mirage/main.ml"
+  # 156 "mirage/main.ml"
   
   let mirage_logs_make__8 = lazy (
     let __pclock__7 = Lazy.force pclock__7 in
@@ -433,17 +445,17 @@ Configure the project for Xen:
     Logs.set_reporter reporter;
     Lwt.return reporter
   );;
-  # 162 "mirage/main.ml"
+  # 168 "mirage/main.ml"
   
   let xen_os_time__9 = lazy (
     return ()
   );;
-  # 167 "mirage/main.ml"
+  # 173 "mirage/main.ml"
   
   let mclock__10 = lazy (
     return ()
   );;
-  # 172 "mirage/main.ml"
+  # 178 "mirage/main.ml"
   
   let mirage_crypto_rng_mirage_make__11 = lazy (
     let __xen_os_time__9 = Lazy.force xen_os_time__9 in
@@ -453,7 +465,7 @@ Configure the project for Xen:
   # 18 "lib/devices/random.ml"
     Mirage_crypto_rng_mirage_make__11.initialize (module Mirage_crypto_rng.Fortuna)
   );;
-  # 182 "mirage/main.ml"
+  # 188 "mirage/main.ml"
   
   let app_make__12 = lazy (
     let __mirage_crypto_rng_mirage_make__11 = Lazy.force mirage_crypto_rng_mirage_make__11 in
@@ -461,7 +473,7 @@ Configure the project for Xen:
   # 3 "config.ml"
     (App_make__12.start _mirage_crypto_rng_mirage_make__11 : unit io)
   );;
-  # 190 "mirage/main.ml"
+  # 196 "mirage/main.ml"
   
   let mirage_runtime__13 = lazy (
     let __struct_end__2 = Lazy.force struct_end__2 in
@@ -481,7 +493,7 @@ Configure the project for Xen:
   # 385 "lib/mirage.ml"
     return ()
   );;
-  # 210 "mirage/main.ml"
+  # 216 "mirage/main.ml"
   
   let () =
     let t = Lazy.force struct_end__2 >>= fun _ ->

